diff --git a/node_modules/@lavanet/lava-sdk/bin/src/badge/badgeManager.js b/node_modules/@lavanet/lava-sdk/bin/src/badge/badgeManager.js
index aae75cf..96939be 100644
--- a/node_modules/@lavanet/lava-sdk/bin/src/badge/badgeManager.js
+++ b/node_modules/@lavanet/lava-sdk/bin/src/badge/badgeManager.js
@@ -35,7 +35,7 @@ class BadgeManager {
         if (options.authentication) {
             this.authentication.append("Authorization", options.authentication);
         }
-        this.transport = transport;
+        this.transport = transport || options.transport;
         this.badgeGeneratorClient = new badges_pb_service_1.BadgeGeneratorClient(this.badgeServerAddress, this.getTransportWrapped());
     }
     isActive() {
diff --git a/node_modules/@lavanet/lava-sdk/bin/src/sdk/sdk.js b/node_modules/@lavanet/lava-sdk/bin/src/sdk/sdk.js
index 89267e8..90f6a88 100644
--- a/node_modules/@lavanet/lava-sdk/bin/src/sdk/sdk.js
+++ b/node_modules/@lavanet/lava-sdk/bin/src/sdk/sdk.js
@@ -89,6 +89,7 @@ class LavaSDK {
             if (!this.badgeManager.isActive() || this.privKey != "") {
                 const wallet = yield (0, wallet_1.createWallet)(this.privKey);
                 this.account = yield wallet.getConsumerAccount();
+                this.walletAddress = this.account.address;
             }
             else {
                 const { wallet, privKey } = yield (0, wallet_1.createDynamicWallet)();
